src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -1816,10 +1824,15 @@ class PostgresAccessPolicyDAO(
     val notNullConstraintRoleAction =
       samsqls"and not (${resourceRole.role} is null and ${resourceAction.action} is null)"
     val notNullConstraintPolicyAction = samsqls"and not (${resourceAction.action} is null)"
+    val resourceParentsConstraint =
+      if (parentResourceIds.nonEmpty) samsqls"and ${parentResourceIds
+          .map(parentResourceId => samsqls"(${resourceParent.resourceTypeId} = ${resourceTypePKsByName(parentResourceId.resourceTypeName)} and ${resourceParent.name} = ${parentResourceId.resourceId})")
+          .reduce((acc, clause) => samsqls"$acc or $clause")}"
+      else samsqls""
```

Comment:
A series of `or` clauses is needed here because we're joining on the fully-qualified resource id, which is actually 2 values.